speech/speech_samples.rb

```diff
@@ -12,160 +12,168 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def speech_sync_recognize project_id: nil, audio_file_path: nil
+def speech_sync_recognize audio_file_path: nil
 # [START speech_sync_recognize]
-  # project_id      = "Your Google Cloud project ID"
   # audio_file_path = "Path to file on which to perform speech recognition"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize
+  audio_file = File.binread audio_file_path
+  config     = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio      = { content: audio_file }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
+
+  alternatives = response.results.first.alternatives
+
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
   end
 # [END speech_sync_recognize]
 end
 
-def speech_sync_recognize_words project_id: nil, audio_file_path: nil
+def speech_sync_recognize_words audio_file_path: nil
 # [START speech_sync_recognize_words]
-  # project_id      = "Your Google Cloud project ID"
   # audio_file_path = "Path to file on which to perform speech recognition"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize words: true
+  audio_file = File.binread audio_file_path
+  config     = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US", enable_word_time_offsets: true }
+  audio      = { content: audio_file }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
+
+  alternatives = response.results.first.alternatives
 
-    result.words.each do |word|
-      puts "Word: #{word.word} #{word.start_time} #{word.end_time}"
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
+
+    alternative.words.each do |word|
+      start_time = word.start_time.seconds + word.start_time.nanos/1000000000.0
```

Comment:
Yuck.

@frankyn What type/value is `word.start_time` in GAPIC only? Versus what was the type/value of `word.start_time` in hand-written?

Reply:
### ⚠️ FRICTION yo

Reply:
Could we get some more details about the changes here? 😄